libs/api/libbpf_map.cpp

```diff
@@ -408,3 +430,97 @@ libbpf_bpf_map_type_str(enum bpf_map_type t)
 
     return _ebpf_map_display_names[t];
 }
+
+typedef struct perf_buffer
+{
+    std::vector<ebpf_map_subscription_t*> subscriptions;
+
+    perf_buffer() : ctx(nullptr), sample_cb(nullptr) {}
+
+    ~perf_buffer()
+    {
+        EBPF_LOG_ENTRY();
+
+        for (auto& sub : subscriptions) {
+            ebpf_map_unsubscribe(sub);
+        }
+
+        subscriptions.clear();
+        EBPF_LOG_EXIT();
+    }
+
+    void* ctx;
+    perf_buffer_sample_fn sample_cb;
+} perf_buffer_t;
+
+int
+perf_buffer_sample_fn_wrapper(void* ctx, int cpu, void* data, uint32_t size)
+{
+    UNREFERENCED_PARAMETER(cpu);
+    perf_buffer_t* perf_buffer = reinterpret_cast<perf_buffer_t*>(ctx);
+    perf_buffer->sample_cb(perf_buffer->ctx, cpu, data, size);
+    return 0;
+}
+
+struct perf_buffer*
+perf_buffer__new(
+    int map_fd,
+    size_t page_cnt,
+    perf_buffer_sample_fn sample_cb,
+    perf_buffer_lost_fn lost_cb,
+    void* ctx,
+    const struct perf_buffer_opts* opts)
+{
+    ebpf_result result = EBPF_SUCCESS;
+    perf_buffer_t* local_perf_buffer = nullptr;
+
+    UNREFERENCED_PARAMETER(opts);
+    UNREFERENCED_PARAMETER(page_cnt);
```

Comment:
Do we need to support page_cnt != 0 or opts != nullptr?  If not, should we fail if not 0/null?